web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -124,17 +126,29 @@ export class MapComponent implements AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
+    this.initMap();
+  }
+
+  ngOnChanges() {
+    this.initMap();
+  }
+
+  initMap() {
     this.subscription.add(
       combineLatest([
         this.activeSurvey$,
         this.lois$,
         this.navigationService.getLocationOfInterestId$(),
       ]).subscribe(([survey, lois, locationOfInterestId]) => {
         const loisMap = ImmutableMap(
-          (this.showPredefinedLoisOnly
-            ? lois.filter(loi => loi.predefined !== false)
-            : lois
-          ).map(loi => [loi.id, loi])
+          lois
+            .filter(loi =>
+              this.showPredefinedLoisOnly ? loi.predefined !== false : loi
```

Comment:
Can `loi` ever be falsey? If so, why not just return `loi => loi.predefined !== false` as before?

Reply:
RIght, I don't know why I wrote it that way, changed.